### PR TITLE
Enable logging support in python sdk#347

### PR DESCRIPTION
**Pull request by dipendughosh**
Enabled logging in python sdk. The framework support for logging is enabled. Need to add more logging as and when needed.

### CODE REVIEW

File: openapiart/common.py [hunk 1]
```diff
@@ -42,6 +54,7 @@ def api(
     loglevel=logging.INFO,
     ext=None,
     version_check=False,
+    module="common",
```
Comment (Vibaswan) [inline openapiart/common.py:57]: Where is the module being used ?

File: openapiart/common.py [hunk 2]
```diff
@@ -835,6 +888,12 @@ def _validate_required(self):
                 raise ValueError(msg)
 
     def _validate_types(self, property_name, property_value):
+        log.info(
```
Comment (Vibaswan) [inline openapiart/common.py:891]: should be a debug I guess

**Review by Vibaswan — CHANGES_REQUESTED** [review]

File: openapiart/common.py [hunk 3]
```diff
@@ -28,6 +28,18 @@
...
 
+# instantiate the logger
+stdout_handler = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter(
+    fmt="%(asctime)s [%(name)s] [%(levelname)s] %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+formatter.converter = time.gmtime
+stdout_handler.setFormatter(formatter)
+log = logging.Logger("common", level=logging.WARN)
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:39]: done

File: openapiart/common.py (hunk 1 above)
Comment (dipendughosh) [reply openapiart/common.py:57]: done

File: openapiart/common.py [hunk 4]
```diff
@@ -71,6 +84,18 @@ class send_recv method to communicate with a REST based server.
     - ext (str): Name of an extension package
     """
     params = locals()
+
+    if logger is not None:
+        global log
+        log = logger
+    else:
+        log.info("Logger already instantiated ")
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:92]: done

File: openapiart/common.py [hunk 5]
```diff
@@ -271,6 +303,7 @@ def serialize(self, encoding=JSON):
             raise NotImplementedError("Encoding %s not supported" % encoding)
         # TODO: restore behavior
         # self._validate_coded()
+        log.warning("Serializeing data - " + str(data))
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:306]: done

File: openapiart/common.py [hunk 6]
```diff
@@ -300,6 +333,7 @@ def deserialize(self, serialized_object):
         self._decode(serialized_object)
         # TODO: restore behavior
         # self._validate_coded()
+        log.warning("Deserializeing")
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:336]: done

File: openapiart/common.py [hunk 7]
```diff
@@ -339,6 +373,7 @@ def validate_mac(self, mac):
         try:
             if len(mac) != 17:
                 return False
+            log.debug("Validating MAC address - " + str(mac))
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:376]: done

File: openapiart/common.py [hunk 8]
```diff
@@ -353,6 +388,7 @@ def validate_ipv4(self, ip):
         if len(ip.split(".")) != 4:
             return False
         try:
+            log.debug("Validating IPv4 address - " + str(ip))
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:391]: done

File: openapiart/common.py [hunk 9]
```diff
@@ -383,6 +419,7 @@ def validate_ipv6(self, ip):
         else:
             ip = ip.replace("::", ":0:")
         try:
+            log.debug("Validating IPv6 address - " + str(ip))
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:422]: done

File: openapiart/common.py [hunk 10]
```diff
@@ -677,6 +726,9 @@ def _get_property(
         return self._properties[name]
 
     def _set_property(self, name, value, choice=None):
+        log.debug(
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:729]: DONE

File: openapiart/common.py (hunk 2 above)
Comment (dipendughosh) [reply openapiart/common.py:891]: done

File: openapiart/common.py [hunk 11]
```diff
@@ -920,6 +979,7 @@ def get(self, name, with_default=False):
         """
         getattr for openapi object
         """
+        log.debug("Get name - " + str(name))
```
[the remark this answers is not in the document]
Comment (dipendughosh) [reply openapiart/common.py:982]: done

**Review by Vibaswan — APPROVED** [review]

**Pull request closed without merging**